### PR TITLE
feat: new SESBulkTemplatedCrudPolicy_v2 policy

### PR DESCRIPTION
**Pull request by aaythapa**
### Issue #, if available

#2844 

### Description of changes

Added a new version of the `SESBulkTemplatedCrudPolicy` policy. `SESBulkTemplatedCrudPolicy_v2` is same but with template arn added to the resource section.

### Description of how you validated changes

Made an example template and manually changed the policy to include the template arn in the resource section. No errors showed.

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [x] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [x] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hoffa — APPROVED** [review]

File: samtranslator/policy_templates_data/policy_templates.json
Comment (ssenchenko) [inline samtranslator/policy_templates_data/policy_templates.json:1923]: I believe we decided that `_v2` should be a new file, not a new policy

**Review by ssenchenko — CHANGES_REQUESTED** [review]

File: samtranslator/policy_templates_data/policy_templates.json
Comment (aaythapa) [reply samtranslator/policy_templates_data/policy_templates.json:1923]: going to double check this

Comment (SimonCMoore) [reply samtranslator/policy_templates_data/policy_templates.json:1923]: hmm actually I thought it was just a new policy with _v2 in the same file rather than a new file. (I was thinking each was in its own file rather than all in one big file, not sure we really want to copy paste the whole thing to add one more policy).  Others should weigh in.

Comment (hoffa) [reply samtranslator/policy_templates_data/policy_templates.json:1923]: This is a regular additive change, I see no benefit to a separate file.

Comment (ssenchenko) [reply samtranslator/policy_templates_data/policy_templates.json:1923]: There is a difference if we set up an automated check for policy templates as we have for profiles. Having all versions in one file will makes running those checks much harder in case an older version is considered problematic, and a new one has a fix and both are in the same file.

Comment (awood45) [reply samtranslator/policy_templates_data/policy_templates.json:1923]: I think that's solvable - same file is going to be a lot simpler, although as we scale some sort of breaking apart (perhaps by logical groupings like service) will be easier to comprehend and to maybe add the possibility of lazy loading if/when performance becomes an issue.

File: samtranslator/policy_templates_data/policy_templates.json
Comment (awood45) [inline samtranslator/policy_templates_data/policy_templates.json:1939]: Have we tested this? Many of these actions do not support a resource constraint, which might cause them to fail if it is present. You may need to divide the actions amongst resource-constrained and `*`-constrained depending on what resource types are supported in this table: https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonses.html

**Review by awood45 — CHANGES_REQUESTED** [review]

File: samtranslator/policy_templates_data/policy_templates.json
Comment (aaythapa) [reply samtranslator/policy_templates_data/policy_templates.json:1939]: Great catch, you were right. The actions `ses:GetIdentityVerificationAttributes` and `ses:VerifyEmailIdentity` don't support resource constraints so those action will fail. They're also part of the `SESBulkTemplatedCrudPolicy` and the `SESCrudPolicy` which both have resource constrains so they'll fail with those policies as well. 

Going to take those actions out of this policy and create a separate, *-constrained, policy for them.

**Review by ssenchenko — APPROVED** [review]
After discussions with @SimonCMoore and @hoffa I think it's okay to have it all in a single plain json object

**Comment by aaythapa** [comment]
Waiting on security approval

**Review by awood45 — APPROVED** [review]

**Pull request merged**
